### PR TITLE
Add more tab validation

### PR DESCRIPTION
**Pull request by CFarcy**

### CODE REVIEW

File: src/modules/RA/DataLayer/components/DataLayerForm.js [hunk 1]
```diff
@@ -19,6 +19,9 @@ const DataLayerForm = ({ ...props }) => {
   const [miniSheetInError, setMiniSheetInError] = React.useState(false);
   const [tableInError, setTableInError] = React.useState(false);
   const [filterInError, setFilterInError] = React.useState(false);
+  const [legendInError, setLegendInError] = React.useState(false);
+  const [styleInError, setStyleInError] = React.useState(false);
+  const [definitionInError, setDefinitionInError] = React.useState(false);
```
Comment (dtrucs) [inline src/modules/RA/DataLayer/components/DataLayerForm.js:24]: To manage all theses errors states, maybe you could refacto all theses useState by an useReducer hook ?

Comment (jrmi) [reply src/modules/RA/DataLayer/components/DataLayerForm.js:24]: Seems indeed a good use case.

Comment (CFarcy) [reply src/modules/RA/DataLayer/components/DataLayerForm.js]: Done.

File: src/modules/RA/DataLayer/components/DataLayerForm.js [hunk 2]
```diff
@@ -13,12 +13,26 @@ import MinisheetTab from './tabs/MinisheetTab';
...
+};
+
+const inErrorReducer = (state, { type, payload }) => {
+  if (state[type] === Boolean(payload)) {
+    return state;
+  }
+  return { ...state, [type]: Boolean(payload) };
+};
+
+const DataLayerForm = React.memo(({ ...props }) => {
```
Comment (dtrucs) [inline src/modules/RA/DataLayer/components/DataLayerForm.js:33]:
```suggestion
const DataLayerForm = React.memo(props => {
```

File: src/modules/RA/DataLayer/components/DataLayerForm.js [hunk 3]
```diff
@@ -13,12 +13,26 @@ import MinisheetTab from './tabs/MinisheetTab';
...
+  minisheet: false,
+  filter: false,
+  table: false,
+};
+
+const inErrorReducer = (state, { type, payload }) => {
+  if (state[type] === Boolean(payload)) {
+    return state;
+  }
+  return { ...state, [type]: Boolean(payload) };
```
Comment (dtrucs) [inline src/modules/RA/DataLayer/components/DataLayerForm.js:30]: Do you really need to spread `state` when setting the `[type]`?

**Review by jrmi — APPROVED** [review]

File: src/modules/RA/DataLayer/components/DataLayerForm.js [hunk 4]
```diff
@@ -13,12 +13,26 @@ import MinisheetTab from './tabs/MinisheetTab';
...
+};
+
+const DataLayerForm = React.memo(({ ...props }) => {
   const [external, setExternal] = React.useState(true);
-  const [popupInError, setPopupInError] = React.useState(false);
-  const [miniSheetInError, setMiniSheetInError] = React.useState(false);
-  const [tableInError, setTableInError] = React.useState(false);
-  const [filterInError, setFilterInError] = React.useState(false);
+  const [errorState, dispatch] = React.useReducer(inErrorReducer, initialErrorState);
```
Comment (dtrucs) [inline src/modules/RA/DataLayer/components/DataLayerForm.js:35]: Greeaaaat! 👍

**Review by dtrucs — CHANGES_REQUESTED** [review]

File: src/modules/RA/DataLayer/components/DataLayerForm.js (hunk 3 above)
Comment (CFarcy) [reply src/modules/RA/DataLayer/components/DataLayerForm.js:30]: I think you have to, or you will lose the `errorState` for the other type (i.e other tabs)

Comment (dtrucs) [reply src/modules/RA/DataLayer/components/DataLayerForm.js:30]: Okay, it looks good to me 👍

**Review by dtrucs — APPROVED** [review]

**Pull request merged**